nuxeo-services/nuxeo-platform-types-core/src/main/java/org/nuxeo/ecm/platform/types/TypeRegistry.java

```diff
@@ -236,4 +245,39 @@ public Type getType(String id) {
         return types.get(id);
     }
 
+    /**
+     * @since 8.4
+     */
+    protected void updateCoreContribution(String id, Type contrib) {
+        SchemaManagerImpl schemaManager = (SchemaManagerImpl) Framework.getService(SchemaManager.class);
+
+        // if there's already a core contribution, unregiser it and register a new one
+        if (dtds.containsKey(id)) {
+            schemaManager.unregisterDocumentType(dtds.get(id));
```

Comment:
Each Type contributed to this non-core TypeService will only contribute a single descriptor to the SchemaManager in append mode, and it is this descriptor that's going to be removed later, not any of the other descriptors, so facets and schemas should not be affected. I will improve the test to cover this aspect.